### PR TITLE
Add MariaDB to CI

### PR DESCRIPTION
**Pull request by robertsilen**
#### Proposed Changes ####

I suggest to add MariaDB to CI testing. 

#### Types of Changes ####

Change to CI test. 

#### User-Facing Change ####

```release-note

```

#### Further Comments ####

I identified where the database type is checked, and added a section for mariadb, but honestly I don't know how to take it further from here. Can I get some tips on how to work on this?

### CODE REVIEW

File: tests/e2e/upgradecluster/Vagrantfile [hunk 1]
```diff
@@ -91,7 +91,17 @@ end
...
+  if EXTERNAL_DB == "mariadb"
+    if role.include?("server") && role_num == 0
+      dockerInstall(vm)
+      vm.provision "Start mysql", inline: "docker run -d -p 3306:3306 --name #{EXTERNAL_DB} -e MYSQL_ROOT_PASSWORD=e2e mysql:5.7"
+      vm.provision "shell", inline: "echo \"Wait for mysql to startup\"; sleep 10"
+      return "datastore-endpoint: 'mysql://root:e2e@tcp(#{NETWORK_PREFIX}.100:3306)/k3s'"
+    elsif role.include?("server") && role_num != 0
+      return "datastore-endpoint: 'mysql://root:e2e@tcp(#{NETWORK_PREFIX}.100:3306)/k3s'"
+    end
+
```
Comment (dereknola) [inline tests/e2e/upgradecluster/Vagrantfile:103]:
```suggestion
    if role.include?("server") && role_num == 0
      dockerInstall(vm)
      vm.provision "Start mariaDB", inline: "docker run -d -p 3306:3306 --name #{EXTERNAL_DB} -e MARIADB_ROOT_PASSWORD=e2e mariadb:11"
      vm.provision "shell", inline: "echo \"Wait for mariaDB to startup\"; sleep 10"
      return "datastore-endpoint: 'mysql://root:e2e@tcp(#{NETWORK_PREFIX}.100:3306)/k3s'"
    elsif role.include?("server") && role_num != 0
      return "datastore-endpoint: 'mysql://root:e2e@tcp(#{NETWORK_PREFIX}.100:3306)/k3s'"
    end
```

Comment (dereknola) [reply tests/e2e/upgradecluster/Vagrantfile:103]: This should get you up and running. The same changes will need to be applies to the validatecluster test. 

Additionally, please sign all your commits to comply with the DCO.

**Review by dereknola — CHANGES_REQUESTED** [review]

**Comment by brandond** [comment]
I will also note that we do validate mariadb on the kine side: https://github.com/k3s-io/kine/blob/master/scripts/test-run-mysql#L28-L30

File: tests/e2e/upgradecluster/Vagrantfile (hunk 1 above)
Comment (robertsilen) [reply tests/e2e/upgradecluster/Vagrantfile]: @dereknola, thanks for help. I believe the commits now comply with DCO

File: tests/e2e/validatecluster/Vagrantfile [hunk 2]
```diff
@@ -99,8 +99,18 @@ def provision(vm, role, role_num, node_num)
...
 
 def getDBType(role, role_num, vm)
-  
-  if EXTERNAL_DB == "mysql"
+
+  if EXTERNAL_DB == "mariadb"
+    if role.include?("server") && role_num == 0
+      dockerInstall(vm)
+      vm.provision "Start mysql", inline: "docker run -d -p 3306:3306 --name #{EXTERNAL_DB} -e MYSQL_ROOT_PASSWORD=e2e mysql:5.7"
+      vm.provision "shell", inline: "echo \"Wait for mysql to startup\"; sleep 10"
```
Comment (dereknola) [inline tests/e2e/validatecluster/Vagrantfile:107]:
```suggestion
      vm.provision "Start mariaDB", inline: "docker run -d -p 3306:3306 --name #{EXTERNAL_DB} -e MARIADB_ROOT_PASSWORD=e2e mariadb:11"
      vm.provision "shell", inline: "echo \"Wait for mariaDB to startup\"; sleep 10"
```

**Review by dereknola — CHANGES_REQUESTED** [review]

**Review by dereknola — APPROVED** [review]

**Review by dereknola — APPROVED** [review]

**Review by brandond — APPROVED** [review]

**Pull request merged**